test/ServiceTest.cpp

```diff
@@ -192,4 +193,34 @@ TEST_F(ServiceTest, computeResult) {
   IdTable expectedIdTable = makeIdTableFromVector(
       {{idX, idY}, {idBla, idBli}, {idBlu, idBla}, {idBli, idBlu}});
   EXPECT_EQ(result->idTable(), expectedIdTable);
+
+  // Check 5: When a siblingTree with variables common to the Service Clause is
+  // passed, the Service Operation shall use the siblings result to reduce
+  // its Query complexity by injecting them as Value Clause
+  auto iri = ad_utility::testing::iri;
+  using TC = TripleComponent;
+  auto siblingTree = std::make_shared<QueryExecutionTree>(
+      testQec,
+      std::make_shared<Values>(
+          testQec,
+          (parsedQuery::SparqlValues){
+              {Variable{"?x"}, Variable{"?y"}, Variable{"?z"}},
+              {{TC(iri("<x>")), TC(iri("<y>")), TC(iri("<z>"))},
+               {TC(iri("<blu>")), TC(iri("<bla>")), TC(iri("<blo>"))}}}));
+
+  auto parsedServiceClause5 = parsedServiceClause;
+  parsedServiceClause5.graphPatternAsString_ = "{ ?x <ble> ?y . }";
+
+  std::string_view expectedSparqlQuery5 =
+      "PREFIX doof: <http://doof.org> SELECT ?x ?y "
+      "WHERE { VALUES ?x { <x> <blu> } . VALUES ?y { <y> <bla> } . "
+      "?x <ble> ?y . }";
```

Comment:
As already stated, this is not quite right (or as efficient as it can be ) yet, but we will get there (see the comment above)